prepare_nic_configs.yml

```diff
@@ -20,35 +20,44 @@
       - block:
         - name: set interfaces
           set_fact:
-            ifaces: "{{ item.value.interfaces }}"
-          with_dict: "{{ scale.rhel7.machine_types.supermicro }}"
-          when: item.key == machine_type and (osp_release <= 13) and machine_type in scale.rhel7.machine_types.supermicro
+            ifaces: "{{ item.value.rhel7_interfaces }}"
+          with_dict: "{{ scale.machine_types.supermicro }}"
+          when: item.key == machine_type and (osp_release <= 13) and machine_type in scale.machine_types.supermicro
 
         - name: set interfaces
           set_fact:
-            ifaces: "{{ item.value.interfaces }}"
-          with_dict: "{{ scale.rhel8.machine_types.supermicro }}"
-          when: item.key == machine_type and (osp_release > 13) and machine_type in scale.rhel8.machine_types.supermicro
+            ifaces: "{{ item.value.rhel8_interfaces }}"
```

Comment:
Try ternary operator here
- name: set interfaces
  set_fact:
      ifaces: "{{ (osp_release|int > 13)| ternary(item.value.rhel8_interfaces, item.value.rhel7_interfaces) }}"
  with_dict: "{{ scale.machine_types.supermicro }}"
  when: item.key == machine_type and machine_type in scale.machine_types.supermicro

With this you can merge two tasks into one.

Reply:
Fixed